examples/chaincode/go/rbac/rbac.go

```diff
@@ -0,0 +1,305 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"encoding/asn1"
+	"github.com/hyperledger/fabric/core/chaincode/shim"
+	"github.com/op/go-logging"
+	"strings"
+)
+
+var myLogger = logging.MustGetLogger("rbac_chaincode")
+
+type RBACMetadata struct {
+	Cert  []byte
+	Sigma []byte
+}
+
+type RBACChaincode struct {
+}
+
+// Init method will be called during deployment
+func (t *RBACChaincode) Init(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {
+	myLogger.Info("Init")
+	if len(args) != 0 {
+		return nil, errors.New("Incorrect number of arguments. Expecting 0")
+	}
+
+	myLogger.Debug("Creating RBAC Table...")
+
+	// Create RBAC table
+	err := stub.CreateTable("RBAC", []*shim.ColumnDefinition{
+		&shim.ColumnDefinition{"ID", shim.ColumnDefinition_BYTES, true},
+		&shim.ColumnDefinition{"Roles", shim.ColumnDefinition_STRING, false},
+	})
+	if err != nil {
+		return nil, errors.New("Failed creating RBAC table.")
+	}
+
+	myLogger.Debug("Assign 'admin' role...")
+
+	// Give to the deployer the role 'admin'
+	deployer, err := stub.GetCallerMetadata()
+	if err != nil {
+		return nil, errors.New("Failed getting metadata.")
+	}
+	if len(deployer) == 0 {
+		return nil, errors.New("Invalid admin certificate. Empty.")
```

Comment:
"Empty.": Maybe "Missing caller metadata in admin certificate", "Deployer information is required in admin certificate", or so?

(trying to be a bit more informative in reporting the error.)